libs/sdk-ui-ext/src/internal/dashboardEmbedding/KpiAlerts/DashboardItemWithKpiAlert.tsx

```diff
@@ -0,0 +1,288 @@
+        }
+
+        if (!this.props.isAlertHighlighted && nextProps.isAlertHighlighted) {
+            this.updateStatePropertyForTime("isAlertHighlighted", 5000);
+        }
+    }
+
+    componentDidUpdate(): void {
+        if (this.props.isAlertHighlighted && !this.isScrolledToHighlightedAlert) {
+            this.isScrolledToHighlightedAlert = true;
+            const node = this.node.current;
+
+            if (node) {
+                window.scrollTo(0, getNodeDocumentRelativeOffsetTop(node));
+            }
+        }
+    }
+
+    componentWillUnmount(): void {
+        this.clearUpdatingTimeout();
+    }
+
+    // toggle property to true for given amount of time
+    updateStatePropertyForTime(name: keyof IDashboardItemWithKpiAlertState, timeout: number): void {
+        const { isKpiAlertAfterSaving, isKpiAlertAfterDeleting, isAlertHighlighted } = this.state;
+
+        this.clearUpdatingTimeout(name);
+
+        this.setState({
+            isKpiAlertAfterSaving,
+            isKpiAlertAfterDeleting,
+            isAlertHighlighted,
+            [name]: true,
+        });
+
+        this.timeouts[name] = setTimeout(() => {
+            this.setState({
+                isKpiAlertAfterSaving,
+                isKpiAlertAfterDeleting,
+                isAlertHighlighted,
+                [name]: false,
+            });
+        }, timeout);
+    }
+
+    clearUpdatingTimeout(name: string = null): void {
+        if (this.timeouts[name]) {
+            clearTimeout(this.timeouts[name]);
+            delete this.timeouts[name];
+        }
+    }
+
+    isKpiAlertSaved(nextProps: IDashboardItemWithKpiAlertProps): boolean {
+        return (
+            !this.state.isKpiAlertAfterSaving &&
+            this.props.alertSavingStatus === "inProgress" &&
+            nextProps.alertSavingStatus === "idle"
+        );
+    }
+
+    isKpiAlertDeleted(nextProps: IDashboardItemWithKpiAlertProps): boolean {
+        return (
+            !this.state.isKpiAlertAfterDeleting &&
+            this.props.alertDeletingStatus === "inProgress" &&
+            nextProps.alertDeletingStatus === "idle"
+        );
+    }
+
+    renderAlertBox = (): React.ReactNode => {
+        const isAlertingTemporarilyDisabled = isAlertingTemporarilyDisabledForGivenFilter(
+            this.props.kpi,
+            this.props.filters,
+            this.props.userWorkspaceSettings,
+        );
+
+        const alertIconClasses = cx(
+            "dash-item-action",
+            "dash-item-action-alert",
+            "s-dash-item-action-alert",
+            "icon-bell",
+            {
+                "alert-set": this.state.isKpiAlertAfterSaving,
+                "alert-deleted": this.state.isKpiAlertAfterDeleting,
+            },
+        );
+
+        // TODO: Remove "isAlertingTemporarilyDisabledForGivenFilter" when alerts support absolute filters (RAIL-1456, RAIL-1457).
+        //       When alert is set, we allow opening the alert box so user can edit/delete it.
```

Comment:
this TODO is here for a while. no need to address now but we should plan do a quick check eventually. we are already using AFM to execute KPIs on the backend so perhaps this is supported

Reply:
On front end maybe, but there is probably some backend code executing this periodically to trigger the alerts, right?

Reply:
even the alerting worker on the backend should be using AFM. imho no action needed now, however a quick followup to verify & if needed address would be in order sometime in near future,